bool initialized_
string data